### DISCUSSION TITLE
Add "toggle" action to settings that only have 2 states (such as "visibility")

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#3505] [Enhancement ideas (and their technical discussions)]
## Description

When users want to change the state of a setting that has two values, they must add extra events to detect if the state was toggled.  

![Screenshot 2021-10-23 150008](https://user-images.githubusercontent.com/8879811/138571645-73cc74b9-8d16-42c3-8103-44bad478be22.png)

## Solution suggested

Add an action to "toggle" the state for all settings that only have two states
- Object visibility (show/hide)
- Layer visibility (show/hide)
- Behaviors (enabled/disabled)
- Others?

**Comment by willianholtz** [comment]
Music/Sound can have too!

see my example to access game inventory.
![Inventoy](https://user-images.githubusercontent.com/23488207/138969463-d2d9c27b-891d-464b-8809-f06f95d35190.png)

**Comment by Bouh** [comment]
Partially done, it's available for boolean variables.
![image](https://user-images.githubusercontent.com/1670670/149673531-2eae0b1f-e50d-4d96-a733-22793ab543c8.png)

As this is a request and not an issue I convert it to a discussion.

**Comment by Elairyx7301** [comment]
The "toggle boolean" method is nice and is easy to implement in the code but I would also plead for toggle actions that Tristan has suggested. Every toggle would save at least two events and conditions per use. In my opinion, the code can get unclear quickly with this since when the events are folded, all that is visible is the name of the boolean and unfolded events will take up much space whereas one toggle action would keep it cleaner.

As far as I'm aware, adding toggles like that is not connected with a lot of performance impacting and possibly even less effort than making toggles manually and will reduce the amount of used scene variable as well.
